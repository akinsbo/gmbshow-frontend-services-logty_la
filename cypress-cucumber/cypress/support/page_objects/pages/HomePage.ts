/**
 * Copyright (c) 2018-present, Mbshow.
 *
 * HeaderFooterPage.js
 * This class defines and exports the key higher order components of the headerfooterPage
 *
 * @author Olaolu Akinsete<akinsbo@gmail.com>, 2018
 *
 */

/**
 * This abstracts common parameters for all pages in the website
 * Current TODO items:
 * X-  Support more keyfeatures in the homepage
 * - Initial test coverage
 * - Support for multiple methods
 */
import makePage from "./BasePage"
import { WebPage, BaseWebPage, FeaturesList, Route } from "../../../types"
import componentFactory from "../ComponentFactory"

export default function makeHomePage(relPath: Route): WebPage {
  const basePage: BaseWebPage = makePage(relPath)

  const componentList: FeaturesList = {
    headerComponent: componentFactory("header")
    // footerComponent: componentFactory('footerComponent'),
    // featuredComponent: componentFactory('featuredComponent'),
    // videoCardComponent: componentFactory('videoCardComponent'),
    // searchComponent: componentFactory('searchComponent'),
    // messageComponent: componentFactory('messageComponent'),
  }

  return Object.freeze({
    basepage: basePage,
    componentList
  })
}
