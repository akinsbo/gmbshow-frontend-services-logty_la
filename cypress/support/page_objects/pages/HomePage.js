/**
 * Copyright (c) 2018-present, Mbshow.
 *
 * HeaderFooterPage.ts
 * This class defines and exports the key higher order components of the headerfooterPage
 *
 * @author Olaolu Akinsete<akinsbo@gmail.com>, 2018
 *
 */

/**
 * This abstracts common parameters for all pages in the website
 * Current TODO items:
 * X-  Support more keyfeatures in the homepage
 * - Initial test coverage
 * - Support for multiple methods
 */
import makePage from "./BasePage"
import componentFactory from "../ComponentFactory"

export default function makeHomePage(relPath) {
  const basePage = makePage(relPath)

  const componentList = {
    headerComponent: componentFactory("header")
    // footerComponent: componentFactory('footerComponent'),
    // featuredComponent: componentFactory('featuredComponent'),
    // videoCardComponent: componentFactory('videoCardComponent'),
    // searchComponent: componentFactory('searchComponent'),
    // messageComponent: componentFactory('messageComponent'),
  }

  return Object.freeze({
    basepage: basePage,
    componentList
  })
}
