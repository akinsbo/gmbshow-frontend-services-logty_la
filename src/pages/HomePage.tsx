import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import * as React from "react"
import Base from '../components/templates/Base'
import HeadModifier from "../core/HeadModifier"
import duyonoStonePng from "../images/duyono-stone.png"
// import withRoot from '../utils/theme/withRoot'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    button: {
      margin: theme.spacing(1),
    },
    grow: {
      flexGrow: 1,
    },
    input: {
      display: 'none',
    },
  })
)

const HomePage = () => {

  const classes = useStyles()
  const { grow } = classes

  return (
    <React.Fragment>
      <div className={`application ${grow}`}>
        <Base />
        <HeadModifier title="Homepage" />
        {/* <Label>Wow</Label> */}
        <h1>This is the Home page</h1>

        <img src={duyonoStonePng} alt="duyonoStonePng" />
      </div>
    </React.Fragment>
  )
}

// export default withRoot(HomePage)
export default HomePage
