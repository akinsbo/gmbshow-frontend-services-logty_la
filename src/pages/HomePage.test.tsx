
import * as Enzyme from "enzyme"
import enzymeAdapterReact16 from "enzyme-adapter-react-16"
import "jest-enzyme"
import * as React from "react"
import { HelmetProvider } from "react-helmet-async"
import * as renderer from "react-test-renderer"
import HomePage from "./HomePage"

Enzyme.configure({
  adapter: new enzymeAdapterReact16()
})

// Reusable homepage and it's wrappers
const homePageAndWrappers =
  <HelmetProvider>
    <HomePage />
  </HelmetProvider>

// We define the wrapper
const wrapper = Enzyme.mount(
  homePageAndWrappers
)

describe("HomePage", () => {
  // Snapshot it
  it("renders correctly", () => {
    const tree = renderer.create(
      homePageAndWrappers
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  describe("structure", () => {
    // console.log(wrapper.debug())

    describe("HeadModifier", () => {
      let headModifierWrapper: any
      beforeEach(() => {
        headModifierWrapper = wrapper.find("HeadModifier")
      })
      it("should be present", () => {
        expect(headModifierWrapper).toHaveLength(1)
      })
      it("should contain props title", () => {
        expect(headModifierWrapper).toHaveProp("title", "Homepage")
      })
    })// describe HeadModifier

  })// describe structure

})// describe HomePage
