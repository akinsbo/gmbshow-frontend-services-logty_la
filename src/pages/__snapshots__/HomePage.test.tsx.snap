// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div
  className="application makeStyles-grow-2"
>
  <div
    className="makeStyles-grow-5"
  >
    <div
      className="header makeStyles-grow-9"
    >
      <header
        className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="Open drawer"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart makeStyles-menuButton-10"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
          </button>
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap makeStyles-titleStyle-12"
          >
            title
          </h6>
          <div
            className="makeStyles-grow-9"
          />
          <div
            className="makeStyles-sectionDesktop-14"
          />
          <div
            className="makeStyles-sectionMobile-15"
          />
        </div>
      </header>
    </div>
  </div>
  <div />
  <h1>
    This is the Home page
  </h1>
  <img
    alt="duyonoStonePng"
    src="test-file-stub"
  />
</div>
`;
